nixos-rebuild: Fix quoting in build-image output, improve docs

phaer opened the pull request:
Don't quote `imageName` in `nixos-rebuild build-image`. It's just a string, not actual json. This mixup - which I authored - leads to extra quoting when printing the image name, e.g.:

```
Done.  The disk image can be found in /nix/store/rynka0dxk4n6c29mggijf8cmbqzf47zh-nixos-image-amazon-25.0
5pre-git-x86_64-linux/"nixos-image-amazon-25.05pre-git-x86_64-linux.vpc"
```

The rest is just documentation updates

phaer committed:
manual: explain per-variant customizations for build-image

phaer committed:
manual: smaller fixes for nixos-rebuild build-image docs

* correct option type for `image.modules`
* fix grammar
* show actual output of the command to

fricklerhandwerk commented on nixos/doc/manual/installation/building-images-via-nixos-rebuild-build-image.chapter.md:
```diff
@@ -2,15 +2,26 @@
+While those can be imported directly, `system.build.images` provides an attribute set mapping variant names to image derivations. Available variants are defined - end extendable - in `image.modules`, an attribute set mapping variant names to NixOS modules.
 
-All of those images can be built via both, their `system.build.image` attribute, and the CLI `nixos-rebuild build-image`. To build i.e. an Amazon image from your existing NixOS configuration:
+All of those images can be built via both, their `system.build.image` attribute, as well as the CLI command `nixos-rebuild build-image`. To build i.e. an Amazon image from your existing NixOS configuration, run:
```
```suggestion
All of those images can be built via both their `system.build.image` attribute, and the `nixos-rebuild build-image` command.

For example, to build an Amazon image from your existing NixOS configuration, run:
```

Although the example should probably be in an `:::{.example}` block.

phaer replied:
I applied it to the block on customizing images below, but I don't think rendering this part as a <detail> which is collapsed by default in the manual is a good idea. It's pretty much at the core of this section, not an extra example or such.

Is there a way to not collapse it?

While looking for examples in `nixos/doc/manual` and grepping for example blocks, I only found a single one in "customising packages".

fricklerhandwerk replied:
> Is there a way to not collapse it?

Yes, change the rendering code. TBH I don't undestand why it was ever made that way. It's the same level of wrong as code comments being low contrast in syntax highlighting: that's very important information, why would you hide it?

fricklerhandwerk replied:
BTW did you run `rg example **/*.md` or what exactly? Because examples typically get an anchor, so it looks more like `:::{.example #something}`.

phaer replied:
I mean, I do love rabbit-holes as much as any other nix users, but please forgive me if I don't take a detour to the rendering code of the manual while trying to add 3 new paragraphs to it :sweat_smile: 

The exact command I ran inside the manual dir was:

```
rg ':::\{.example'
```

which did catch blocks with anchors.

fricklerhandwerk replied:
>  I do love rabbit-holes as much as any other nix users

Yeah I didn't mean you should do it now, but answered the question what can be done :) 

> While looking for examples in `nixos/doc/manual` and grepping for example blocks, I only found a single one in "customising packages".

Ah, it's just NixOS docs, not Nixpkgs. Yes, possible that it's just not particularly well maintained. If people complain "Nix documentation sucks" what they really mean is "NixOS documentation sucks", which is no surprise if barely anyone is working on it.

fricklerhandwerk commented on nixos/doc/manual/installation/building-images-via-nixos-rebuild-build-image.chapter.md:
```suggestion
The `image.modules` option can be used to set specific options per image variant, in a similar fashion as [specialisations](options.html#opt-specialisation) for generic NixOS configurations.
```

phaer replied:
thanks

fricklerhandwerk commented on nixos/doc/manual/installation/building-images-via-nixos-rebuild-build-image.chapter.md:
```suggestion
E.g. images for the cloud provider Linode use `grub2` as a bootloader by default. If you are using `systemd-boot` on other platforms and want to disable it for Linode only, you could use the following options:
```

Again, this should be an example block.

phaer committed:
nixos-rebuild: don't quote imageName for build-image

it's just a string, not actual json. This lead to extra quoting when
printing the image name, i.e.:

Done.  The disk image can be found in /nix/store/rynka0dxk4n6c29mggijf8cmbqzf47zh-nixos-image-amazon-25.0
5pre-git-x86_64-linux/"nixos-image-amazon-25.05pre-git-x86_64-linux.vpc"

phaer commented:
I hope to have addressed all outstanding feedback, so I think this might be good to merge?

spiage commented:
broken nixpkgs-rebuild, line 868, "   | jq -r . "
```
        if [[ -z $buildingAttribute ]]; then
            imageName="$(
                runCmd nix-instantiate --eval --strict --json --expr \
                "let
                    value = import \"$(realpath $buildFile)\";
                    set = if builtins.isFunction value then value {} else value;
                in set.${attr:+$attr.}config.system.build.images.$imageVariant.v.passthru.filePath" \
                "${extraBuildFlags[@]}"
        | jq -r .
            )"
``` 
```
❯ nixos-rebuild -v -L switch --flake .
$ cat /proc/sys/kernel/hostname
$ nix --extra-experimental-features nix-command flakes build --out-link /tmp/nixos-rebuild.FQN5AJ/nixos-rebuild .#nixosConfigurations."a7".config.system.build.nixos-rebuild -v -L
$ exec /nix/store/f2zrxs682q0ql7d3llbkgp16580mzac7-nixos-rebuild/bin/nixos-rebuild -v -L switch --flake .
$ cat /proc/sys/kernel/hostname
/nix/store/f2zrxs682q0ql7d3llbkgp16580mzac7-nixos-rebuild/bin/nixos-rebuild: line 868: syntax error near unexpected token `|'
```

spiage commented:
@thiagokokada @phaer @fricklerhandwerk @jsierles
